src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.cs

```diff
@@ -186,28 +169,28 @@ public static void TimeoutTest_SendTimeout5Seconds()
     [OuterLoop]
     public static void TimeoutTest_SendTimeout0Seconds()
     {
-        bool exceptionThrown = false;
+        BasicHttpBinding binding = new BasicHttpBinding();
+        binding.SendTimeout = TimeSpan.FromMilliseconds(0);
+        ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+
+        Stopwatch watch = new Stopwatch();
         try
         {
-            BasicHttpBinding binding = new BasicHttpBinding();
-            binding.SendTimeout = TimeSpan.FromMilliseconds(0);
-            ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
-            IWcfService proxy = factory.CreateChannel();
-            proxy.EchoWithTimeout("Hello");
-        }
-        catch (TimeoutException)
-        {
-            exceptionThrown = true;
+            var exception = Assert.Throws<TimeoutException>(() =>
+            {
+                IWcfService proxy = factory.CreateChannel();
+                watch.Start();
+                proxy.EchoWithTimeout("Hello");
+            });
         }
-        catch (Exception ex)
+        finally
         {
-            Assert.True(false, String.Format("Unexpected exception caught: {0}", ex.ToString()));
+            watch.Stop(); 
         }
 
-        if (!exceptionThrown)
-        {
-            Assert.True(false, "Expected TimeoutException was not thrown nor was any other exception thrown.");
-        }
+        // want to assert that this completed in < 0.5 s as an upper bound since the SendTimeout is 0 sec
+        // (usual case is around 1 - 3 ms) 
+        Assert.InRange<long>(watch.ElapsedMilliseconds, 0, 500);
```

Comment:
I'm concerned about the InRange here too.  Predicting execution times of lab machines usually makes tests break periodically.  I think it would be okay to simply verify the time is not greater than the default SendTimeout (i.e. verify we did timeout "sooner" than the normal timeout).